server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/UploadModule.java

```diff
@@ -47,11 +48,12 @@ public interface UploadModule {
                 .withCompactionWindow(7, DAYS))
             .withCaching(true, rows(DEFAULT_CACHED_ROW_PER_PARTITION)))
         .statement(statement -> types -> statement
-            .withPartitionKey(ID, DataTypes.TIMEUUID)
+            .withPartitionKey(USER, DataTypes.TEXT)
+            .withClusteringColumn(ID, DataTypes.TIMEUUID)
             .withColumn(CONTENT_TYPE, DataTypes.TEXT)
             .withColumn(SIZE, DataTypes.BIGINT)
             .withColumn(BUCKET_ID, DataTypes.TEXT)
             .withColumn(BLOB_ID, DataTypes.TEXT)
-            .withColumn(USER, DataTypes.TEXT))
+            .withColumn(UPLOAD_DATE, DataTypes.TIMESTAMP))
```

Comment:
Please change the table name: `String TABLE_NAME = "uploads_2";`

Please indicate in upgrade instructions that the `upload` table can be deleted. That pending uploads will be lost during the migration.